examples/delete_disks.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Show how to delete many disks created by the 'add_disks.py' example quickly.
+"""
+
+import time
+from contextlib import closing
+
+from ovirtsdk4 import types
+
+from helpers import common
+from helpers.common import progress
+from helpers.units import humansize, GiB
+
+parser = common.ArgumentParser(description="Delete disks created by the 'add_disks.py' example")
+
+parser.add_argument(
+    "--sd-name",
+    required=True,
+    help="Name of the storage domain.")
+
+args = parser.parse_args()
+
+def get_storage_domain_disk_service_by_desc(sd_service, disk_desc):
+    disks_service = sd_service.disks_service()
+    # StorageDomainDisksService.list has no 'search' parameter and ignores
+    # query={'name': 'spam'} so we have to do the filtering ourselves
+    disk = next((disk for disk in disks_service.list() if disk.description == disk_desc), None)
+    if disk is None:
+        return None
+    else:
+        return disks_service.disk_service(disk.id)
```

Comment:
This is very inefficient - for every disk you list all disks in the system, filter the disks with the wanted description, and return the first found disk.

This is likely fast enough since deleting a disk is so slow, but not good enough
for example code.

If we don't have a way to search the description, the way to find the disks and delete them is:

    disks = [disk for disk in disks_service.list() if disk.description == disk_desc]
    for disk in disks:
        delete it...

Reply:
we don't have a way to search the desc, this is why the mechanism was written that way :)
so you suggest going through the whole list of disks instead of finding them.
right attitude (didn't think about that too much as only wanted to delete the disks from my system, but you're right).